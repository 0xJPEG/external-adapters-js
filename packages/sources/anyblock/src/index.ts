import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig } from './config'

const NAME = 'ANYBLOCK'

const { server } = expose(NAME, makeExecute(), undefined, endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server }
