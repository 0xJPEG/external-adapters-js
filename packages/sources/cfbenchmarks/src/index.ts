import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute, makeWSHandler } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig, NAME } from './config'

const { server } = expose(NAME, makeExecute(), makeWSHandler(), endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server }
