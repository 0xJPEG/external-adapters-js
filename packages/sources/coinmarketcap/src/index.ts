import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute, endpointSelector } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig, NAME } from './config'

const server = expose(NAME, makeExecute(), undefined, endpointSelector).server

export { NAME, endpoints, makeConfig, makeExecute, server }
