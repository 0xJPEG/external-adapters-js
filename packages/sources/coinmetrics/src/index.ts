import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute, makeWSHandler } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig, NAME } from './config'
import * as types from './endpoint'

const { server } = expose(NAME, makeExecute(), makeWSHandler(), endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server, types }
