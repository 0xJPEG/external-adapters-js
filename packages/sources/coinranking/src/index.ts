import { expose } from '@chainlink/ea-bootstrap'
import { makeConfig, NAME } from './config'
import * as endpoints from './endpoint'
import { endpointSelector, makeExecute } from './adapter'

const { server } = expose(NAME, makeExecute(), undefined, endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server }
