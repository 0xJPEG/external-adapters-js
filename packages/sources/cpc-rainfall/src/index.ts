import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute, CALLBACK_PROPERTIES } from './adapter'
import { makeConfig, NAME } from './config'

export = { NAME, makeExecute, makeConfig, ...expose(makeExecute(), undefined, CALLBACK_PROPERTIES) }
