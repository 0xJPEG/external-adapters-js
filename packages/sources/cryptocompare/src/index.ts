import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute, endpointSelector, makeWSHandler } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig, NAME } from './config'

const { server } = expose(NAME, makeExecute(), makeWSHandler(), endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server }
