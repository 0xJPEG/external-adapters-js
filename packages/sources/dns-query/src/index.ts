import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { makeConfig } from './config'

const NAME = 'DNS_Query'

const { server } = expose(NAME, makeExecute())

export { NAME, makeConfig, makeExecute, server }
