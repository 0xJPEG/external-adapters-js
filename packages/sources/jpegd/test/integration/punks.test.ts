import { AdapterRequest } from '@chainlink/types'
import request, { SuperTest, Test } from 'supertest'
import process from 'process'
import nock from 'nock'
import http from 'http'
import { server as startServer } from '../../src'
import { mockPunksValueAPIResponseSuccess, mockPunksValueResponseSuccess } from './fixtures'
import { AddressInfo } from 'net'

describe('execute', () => {
  const id = '1'
  let server: http.Server
  let req: SuperTest<Test>

  beforeAll(async () => {
    process.env.API_KEY = process.env.API_KEY || 'test-key'
    if (process.env.RECORD) {
      nock.recorder.rec()
    }
    server = await startServer()
    req = request(`localhost:${(server.address() as AddressInfo).port}`)
  })

  afterAll((done) => {
    if (process.env.RECORD) {
      nock.recorder.play()
    }

    nock.restore()
    nock.cleanAll()
    nock.enableNetConnect()
    server.close(done)
  })

  describe('punk valuation api', () => {
    const data: AdapterRequest = {
      id,
      data: { block: 'latest' },
    }

    it('should return success', async () => {
      mockPunksValueAPIResponseSuccess()
      mockPunksValueResponseSuccess()

      const response = await req
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
