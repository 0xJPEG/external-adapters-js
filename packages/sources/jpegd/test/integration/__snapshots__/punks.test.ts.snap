// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute punk valuation api should return success 1`] = `
Object {
  "data": Object {
    "result": 71.71525868055556,
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": 71.71525868055556,
  "statusCode": 200,
}
`;
